flake.nix

```diff
@@ -2,177 +2,157 @@
   description = "Nix flake for zkEVM framework";
 
   inputs = {
-    nixpkgs = { url = "github:NixOS/nixpkgs/nixos-23.11"; };
-    nil_evm_assigner = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-3rdparty = {
+      url = "github:NilFoundation/nix-3rdparty";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+    nil-evm-assigner = {
       type = "github";
       owner = "NilFoundation";
       repo = "evm-assigner";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        nil_crypto3.follows = "nil-crypto3";
+        nil_zkllvm_blueprint.follows = "nil-zkllvm-blueprint";
+      };
     };
-    nil_crypto3 = {
+    nil-crypto3 = {
       url = "https://github.com/NilFoundation/crypto3";
       type = "git";
       submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
     };
-    nil_zkllvm_blueprint = {
+    nil-zkllvm-blueprint = {
       url = "https://github.com/NilFoundation/zkllvm-blueprint";
       type = "git";
       submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        nil_crypto3.follows = "nil-crypto3";
+      };
     };
-    intx = { url = "github:chfast/intx"; flake = false; };
-    hashtree = { url = "github:prysmaticlabs/hashtree"; flake = false; };
-    sszpp = { url = "github:OffchainLabs/sszpp"; flake = false; };
-    evmc = { url = "github:ethereum/evmc"; flake = false; };
-    valijson = {url = "github:tristanpenman/valijson"; flake = false; };
   };
 
-  outputs = { self, nixpkgs, nil_evm_assigner, nil_crypto3, nil_zkllvm_blueprint, ... } @ repos:
+  outputs =
+    { self
+    , nixpkgs
+    , flake-utils
+    , nix-3rdparty
+    , nil-evm-assigner
+    , nil-crypto3
+    , nil-zkllvm-blueprint
+    }:
+    flake-utils.lib.eachDefaultSystem (system:
```

Comment:
JFYI when you're trying to search for `flake-utils`, there are a bunch of arguments on it. Here is an [example](https://discourse.nixos.org/t/what-are-reasons-to-not-use-flake-utils/21140)

Reply:
Well, we usually don't pay a lot of attention to the supported systems list, right?

Reply:
Yes, unfortunately, but someone may conclude from `nix flake show` that we support 4 systems. And this fact is arguable right now )))

Reply:
I have good feeling about it, let's check it on ARM!